File: backend/models/boarding_house.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type BoardingHouse struct {
	ID               uint           `gorm:"primaryKey" json:"id"`
	OwnerID          uint           `json:"owner_id"`
	Owner            User           `gorm:"foreignKey:OwnerID" json:"owner"`
	Name             string         `gorm:"size:100" json:"name"`
	Address          string         `gorm:"type:text" json:"address"`
	ImageUrl         string         `gorm:"type:text" json:"image_url"`
	ImageUrls        StringList     `gorm:"type:text" json:"image_urls"`
	DefaultRoomPrice float64        `json:"default_room_price"`
	Rooms            []Room         `json:"rooms"`
	CreatedAt        time.Time      `json:"created_at"`
	UpdatedAt        time.Time      `json:"updated_at"`
	DeletedAt        gorm.DeletedAt `gorm:"index" json:"-"`
}

File: rumatenant/lib/models/boarding_house.dart
class BoardingHouse {
  final int id;
  final int ownerId;
  final String name;
  final String address;
  final String imageUrl;
  final List<String> imageUrls;
  final double defaultRoomPrice;

  BoardingHouse({
    required this.id,
    this.ownerId = 0,
    this.name = '',
    this.address = '',
    this.imageUrl = '',
    this.imageUrls = const [],
    this.defaultRoomPrice = 0,
  });

  factory BoardingHouse.fromJson(Map<String, dynamic> json) {
    return BoardingHouse(
      id: json['id'] ?? 0,
      ownerId: json['owner_id'] ?? 0,
      name: json['name'] ?? '',
      address: json['address'] ?? '',
      imageUrl: json['image_url'] ?? '',
      imageUrls: json['image_urls'] != null
          ? List<String>.from(json['image_urls'])
          : [],
      defaultRoomPrice: (json['default_room_price'] ?? 0).toDouble(),
    );
  }
}
